File: LeetCode/3/main.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length();
        int left = 0;
        int ans = 0;
        vector<bool> vis(256, false); 

        for(int right = 0; right < n; ++right) {
            while(vis[s[right]]) {
                vis[s[left]] = false;
                left++;
            }
            vis[s[right]] = true;
            ans = max(ans, right - left + 1);
        }

        return ans;
    }
};

File: leetcode/3/main.go
package main

func lengthOfLongestSubstring(s string) int {
	n := len(s)
	vis := [256]bool{false}
	left, ans := 0, 0

	for right := 0; right < n; right++ {
		for vis[s[right]] {
			vis[s[left]] = false
			left++
		}
		vis[s[right]] = true
		ans = max(ans, right-left+1)
	}

	return ans
}
